SDK seems to have a race condition on watch registration

When we start a continuous graph with many watcahbles to log, there is often a error reported by the SDK where the server sent a value update before the client was ready for it. Most likely some kind of race condition, to be investigated

Example of log when using the GUI and running a graph

> $ python -m scrutiny gui --auto-connect
> [INFO] <ScrutinyClient> Connected to a Scrutiny server at localhost:8765
> [INFO] <ScrutinyClient> A new device is connected and ready. Session ID: 08b2e45d7d85418a848435faf5ef17ad 
> [INFO] <ScrutinyClient> Server has loaded a Firmware Decription with firmware ID: 71c3631628caffd4f01791828a3cb016
> [ERROR] <ScrutinyClient> Got watchable update for unknown watchable 306
> [INFO] <ScrutinyClient> Device is gone. Last session ID: 08b2e45d7d85418a848435faf5ef17ad
> [INFO] <ScrutinyClient> Server has unloaded a Firmware Decription with firmware ID: 71c3631628caffd4f01791828a3cb016
> [INFO] <ScrutinyClient> Disconnected from server at localhost:8765

--- a/scrutiny/sdk/client.py
+++ b/scrutiny/sdk/client.py
@@ -669,7 +669,7 @@ class ScrutinyClient:
                     watchable = self._watchable_storage[update.server_id]
 
             if watchable is None:
-                self._logger.error(f"Got watchable update for unknown watchable {update.server_id}")
+                self._logger.error(f"Got watchable update for unknown watchable. Server ID={update.server_id}")
                 continue
             else:
                 if self._logger.isEnabledFor(DUMPDATA_LOGLEVEL):   # prgama: no cover
@@ -1441,6 +1441,10 @@ class ScrutinyClient:
                         f'The server did not confirm the subscription for the right watchable. Got {list(response["subscribed"].keys())[0]}, expected {path}')
 
                 watchable._configure(watchable_defs[path])
+                assert watchable._configuration is not None
+                with self._main_lock:
+                    self._watchable_path_to_id_map[watchable.display_path] = watchable._configuration.server_id
+                    self._watchable_storage[watchable._configuration.server_id] = watchable
 
         req = self._make_request(API.Command.Client2Api.SUBSCRIBE_WATCHABLE, {
             'watchables': [watchable.display_path]  # Single element
@@ -1453,12 +1457,10 @@ class ScrutinyClient:
             raise sdk.exceptions.OperationFailure(f"Failed to subscribe to the watchable. {future.error_str}")
 
         watchable._assert_configured()
-        assert watchable._configuration is not None
-        with self._main_lock:
-            self._watchable_path_to_id_map[watchable.display_path] = watchable._configuration.server_id
-            self._watchable_storage[watchable._configuration.server_id] = watchable
+        assert watchable._configuration is not None # To please mypy
+        
         if self._logger.isEnabledFor(logging.DEBUG):
-            self._logger.debug(f"Now watching {watchable.display_path}")
+            self._logger.debug(f"Now watching {watchable.display_path} (Server ID={watchable._configuration.server_id})")
 
         return watchable
 

